packages/image_picker/ios/Classes/ImagePickerPlugin.m

```diff
@@ -110,20 +142,36 @@ - (void)imagePickerController:(UIImagePickerController *)picker
     image = [self scaledImage:image maxWidth:maxWidth maxHeight:maxHeight];
   }
 
-  NSData *data = UIImageJPEGRepresentation(image, 1.0);
-  NSString *tmpDirectory = NSTemporaryDirectory();
-  NSString *guid = [[NSProcessInfo processInfo] globallyUniqueString];
-  // TODO(jackson): Using the cache directory might be better than temporary
-  // directory.
-  NSString *tmpFile = [NSString stringWithFormat:@"image_picker_%@.jpg", guid];
-  NSString *tmpPath = [tmpDirectory stringByAppendingPathComponent:tmpFile];
-  if ([[NSFileManager defaultManager] createFileAtPath:tmpPath contents:data attributes:nil]) {
-    _result(tmpPath);
+  if (videoURL != nil) {
```

Comment:
Could we avoid working on the image above, if `videoURL` is not `nil`? Seems to be wasted effort.